### ISSUE TITLE
FIMS Developer Handbook, GitHub Flow vs trunk-based

### ISSUE THREAD

**Issue by msupernaw** [NOAA-FIMS/collaborative_workflow#19]
https://noaa-fims.github.io/collaborative_workflow/contributor-guidelines.html#fims-branching-strategy

These are actually 2 different things. We want to use trunk-based development.

Description:
https://reviewpad.com/blog/github-flow-trunk-based-development-and-code-reviews/

**Comment by kristanblackhart-NOAA** [comment]
Thanks @msupernaw  I'll fix the text in the dev guide later today.

**Comment by k-doering-NOAA** [comment]
@msupernaw , thanks for pointing this out and the link to the article, it was laid out really well! 

Which trunk-based that is described in the article do you think makes sense for FIMS? The one for smaller or larger teams?

I'm also curious about how well these would work, given that FIMS implementation team members might not always be able to finish their feature quickly. The note about keeping branches for the Scaled Trunk-Based Development "for a max of a few days" doesn't always seem feasible when people can't devote all of their time in a week to development. What are your thoughts about this? Is it a concern?

**Comment by kristanblackhart-NOAA** [comment]
@k-doering-NOAA @msupernaw Let's talk about this with the group during Wednesday's meeting. I'm thinking this would be require some adjustment for most of the team, but keeping to small commits and short-lived branches is the only way to get the benefits of this approach. I guess it's a question of whether the work can be chunked out into small enough bits to make this workable?

**Comment by msupernaw** [comment]
@k-doering-NOAA you've brought up good points. Given the fact that we'll have a lot of junior developers on the team, that all have other commitments, perhaps Git Flow will work best for us! Pull requests give us the opportunity for code review. I have officially reversed my position!

**Comment by kristanblackhart-NOAA** [comment]
@msupernaw what about branch management and merge conflicts with Git Flow? Other strategies allow for code review as well.

**Comment by k-doering-NOAA** [comment]
It's so silly that all these strategies have similar names! In my experience with GitHUB flow, merge conflicts sometimes happen, but are almost always easy to solve. This might be because the number of contributors on projects I've worked on is fairly small, though. Merge conflicts can be minimized by making sure different people or groups of people are working on different modules of the code (and probably in different modules). I'm not sure if branch conflicts are more of an issues in Git Flow, admittedly.

Also, I agree the branch management in Git Flow is a beast. That's why we don't use it for stock synthesis and stick to Github flow. Git flow branch management is just beyond our team's git abilities to deal with. We definitely miss out on some potential benefits, though (e.g., we don't have a good way to hot fix a release; rather, a "fix" goes out with whatever new features we have developed for the next release). 

My last point is of course we can pick one of these and revise our workflow as needed; I think starting simpler would be better, though, given our wide range of git abilities among the team.
